adapter/internal/ga/client_test.go

```diff
@@ -91,23 +96,39 @@ func TestFetchAPIsFromGA(t *testing.T) {
 		Resources: pbArrayStartup,
 	}
 	isFirstResponse = true
-	addAPIToChannel(testDiscoveryResponseStartup)
-	// Test FetchAPIsFromGA() function
-	startupAPIEventsArray := FetchAPIsFromGA()
+	os.Setenv("FEATURE_ENV_BASED_FILTERING_IN_STARTUP", "true")
 
-	// Check Startup API event array
-	for _, startupAPIEvent := range startupAPIEventsArray {
-		if _, ok := apiListMap[startupAPIEvent.APIUUID]; ok {
-			// Check API UUID of the API event
-			assert.Equal(t, startupAPIEvent.APIUUID, apiListMap[startupAPIEvent.APIUUID].ApiUUID,
-				"API UUID should be the same")
-			// Check Revision UUID of the API event
-			assert.Equal(t, startupAPIEvent.RevisionUUID, apiListMap[startupAPIEvent.APIUUID].RevisionUUID,
-				"Revision UUID should be the same")
-			// Check Organization UUID of the API event
-			assert.Equal(t, startupAPIEvent.OrganizationUUID, apiListMap[startupAPIEvent.APIUUID].OrganizationUUID,
-				"Organization UUID should be the same")
+	// Create a channel to receive results from goroutine
+	resultChan := make(chan bool)
+
+	// Process API events in a goroutine
+	go func() {
+		startupAPIEventsArray, startupAPIEnvMap := FetchAPIsFromGA()
+		assert.NotNil(t, startupAPIEnvMap)
+		// Check Startup API event array
+		for _, startupAPIEvent := range startupAPIEventsArray {
+			if _, ok := apiListMap[startupAPIEvent.APIUUID]; ok {
+				// Check API UUID of the API event
+				assert.Equal(t, startupAPIEvent.APIUUID, apiListMap[startupAPIEvent.APIUUID].ApiUUID,
+					"API UUID should be the same")
+				// Check Revision UUID of the API event
+				assert.Equal(t, startupAPIEvent.RevisionUUID, apiListMap[startupAPIEvent.APIUUID].RevisionUUID,
+					"Revision UUID should be the same")
+				// Check Organization UUID of the API event
+				assert.Equal(t, startupAPIEvent.OrganizationUUID, apiListMap[startupAPIEvent.APIUUID].OrganizationUUID,
+					"Organization UUID should be the same")
+				envMap, found := startupAPIEnvMap[startupAPIEvent.APIUUID]
+				assert.True(t, found, "API Entry is not found")
+				apiDeployment, deploymentFound := envMap[startupAPIEvent.DeployedEnv]
```

Comment:
Here we are not atually testing the filtering logic of deployments got from APIM and list of deployments from GA right?